browse/services/documents/db_implementation/db_abs.py

```diff
@@ -92,32 +95,34 @@ def _to_docmeta(all_versions: List[Metadata], latest: Metadata, ver_of_interest:
         entry = VersionEntry(version=ver.version,
                              raw='',
                              size_kilobytes=size_kilobytes,
-                             submitted_date=ver.created.replace(tzinfo=timezone.utc),  # verified as UTC in DB
-                             source_flag=SourceFlag(ver.source_flags),
+                             submitted_date=ver.created.replace(tzinfo=timezone.utc),  # type: ignore
+                             # ^verified as UTC in DB
+                             source_flag=SourceFlag(ver.source_flags), # type: ignore
                              source_format=ver.source_format,
-                             is_withdrawn=ver.is_withdrawn or ver.source_format == "withdrawn"
+                             is_withdrawn=bool(ver.is_withdrawn) or ver.source_format == "withdrawn"
                                           or ver.source_size == 0,
                              is_current= ver.version == len(all_versions))
         version_history.append(entry)
 
     doc_license: License = License() if not ver_of_interest.license else License(recorded_uri=ver_of_interest.license)
 
     modified = ver_of_interest.updated or ver_of_interest.created
-    modified = modified.replace(tzinfo=timezone.utc)  # Verified as UTC in DB
+    modified = modified.replace(tzinfo=timezone.utc) # type: ignore
+    # ^verified as UTC in DB
     primary_category, secondary_categories, primary_archive = _classification_for_metadata(identifier, latest)
 
     this_version = DocMetadata(
         raw_safe='',
-        abstract=ver_of_interest.abstract,
+        abstract=ver_of_interest.abstract, # type: ignore
```

Comment:
What causes these type ignores? Are these cases where the DB can be null but the model is not Optional?

Reply:
Yes exactly